src/backend/storeManagers/legendary/commands/base.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod'
+import path from 'path'
+import { hasGame } from '../library'
+
+export const LegendaryAppName = z
+  .string()
+  .refine((val) => hasGame(val), {
+    message: 'AppName was not found on account'
+  })
+  .brand('LegendaryAppName')
+export type LegendaryAppName = z.infer<typeof LegendaryAppName>
+
+export const LegendaryPlatform = z.enum(['Win32', 'Windows', 'Mac'] as const)
+export type LegendaryPlatform = z.infer<typeof LegendaryPlatform>
```

Comment:
do we need this LegendaryPlatform type using zod? can't this be a normal typescript type?

Not sure how zod works so maybe this is needed to make this work, but we already have `export type LegendaryInstallPlatform = 'Windows' | 'Win32' | 'Mac'` in src/common/types/legendary.ts

Reply:
Zod validates data at runtime, types don't exist anymore at that point
We *could* remove the definition from `common/types/legendary.ts` (& only import the one defined here), but I'm pretty sure we'd then have to switch to using `import type` everywhere (since Vite can't know that you *really* only need the type from this file, imports can have side effects after all)

Reply:
I'm fine with keeping the 2 definitions, it's not an issue